7 vote on article

Jord000 opened the pull request:
adds a voting button to the individual article page. Component is made to be reusable across any voting system. Extra navigation detail added to allow moving back to the entry page if no one is logged in or tries to access an endpoint without logging in.

johnhutcheon commented on MVC/controllers.js:
```diff
@@ -33,10 +35,26 @@ exports.getArticleById = (req, res, next) => {
+  Promise.all(promisesInput)
+    .then((results) => {
+      res.status(200).send({ comments: results[1] })
+    })
```
Very nicely done! 🏅

johnhutcheon commented on __tests__/app.test.js:
```diff
@@ -139,3 +139,44 @@ describe('GET:200 /api/articles', () => {
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: expect.any(Number)
```
We know the article ID here- so we can expect the specific number 👍

johnhutcheon commented on __tests__/app.test.js:
```diff
@@ -139,3 +139,44 @@ describe('GET:200 /api/articles', () => {
+    .then(({ body }) => {
+      expect(body.msg).toEqual('Not Found');
+    });
+  });
```
Can we have another error test here, for when we pass an invalid ID, i.e. articles/banana/comments?

johnhutcheon reviewed (changes requested):
Hey Jordan, nice work! Could you just see my comments and resubmit with the requested test please? Thanks!

nickkagame reviewed (approved):
All good jordan!

Jord000 committed:
set up for comments on article page

Jord000 committed:
comments added by article

Jord000 committed:
error removed

Jord000 committed:
Merge branch '5-individual-article' into 6-comments-to-article

Jord000 committed:
merging branch 5

Jord000 committed:
add upvote and downvote buttons added to article in detail. added navigation to home if no one is logged in

Jord000 committed:
username persists per session now to improve log-in and refresh info
Votes now can only be incremented 1up or 1down from the original value each way - similar to how reddit upvote works

Jord000 committed:
console.log deleted

nc-nick-jones reviewed (approved):
nice changes jordan.  I would also add, that some user feedback if there is an error on the votes (e.g. votes didn't work because you were offline) would be good.  Please add this when you can.

Jord000 committed:
add error handler for voting buttons
